src/System.Diagnostics.EventLog/src/System/Diagnostics/SafeEventLogReadHandle.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    internal sealed class SafeEventLogReadHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        internal SafeEventLogReadHandle() : base(true) { }
+
+        [DllImport(Interop.Libraries.Advapi32, CharSet = System.Runtime.InteropServices.CharSet.Unicode, SetLastError = true)]
+        internal static extern SafeEventLogReadHandle OpenEventLog(string UNCServerName, string sourceName);
```

Comment:
@danmosemsft should the PInvokes here be moved to Comon/Interop? I think the guidelines state so.

Reply:
Any update ?

Reply:
I may have replied elsewhere. Please leave the ones specific to EventLog in the EventLog folder, and any ones of general interest in the Common folder.

Reply:
> I suggested if they are EventLog specific they remain here since they're not going to be used by anyone else.

from Dan.